tests/api/test_app.py

```diff
@@ -116,7 +216,55 @@ def test_app_update_mutation(
             PermissionEnum.MANAGE_USERS.name,
         ],
     }
-    response = staff_api_client.post_graphql(query, variables=variables)
+    response = staff_api_client.post_graphql(
+        query, variables=variables, permissions=(permission_manage_apps,)
+    )
+    content = get_graphql_content(response)
+
+    app_data = content["data"]["appUpdate"]["app"]
+    tokens_data = app_data["tokens"]
+    app.refresh_from_db()
+    tokens = app.tokens.all()
+
+    assert app_data["isActive"] == app.is_active
+    assert app.is_active is False
+    assert len(tokens_data) == 1
+    assert tokens_data[0]["authToken"] == tokens.get().auth_token[-4:]
+    assert set(app.permissions.all()) == {
+        permission_manage_products,
+        permission_manage_users,
+    }
+
+
+def test_app_update_mutation_for_app(
+    permission_manage_apps,
+    permission_manage_products,
+    permission_manage_orders,
+    permission_manage_users,
+    app_api_client,
+):
+    app = App.objects.create(name="New_sa")
+    AppToken.objects.create(app=app)
+    query = APP_UPDATE_MUTATION
+    requestor = app_api_client.app
+    requestor.permissions.add(
+        permission_manage_apps,
+        permission_manage_products,
+        permission_manage_users,
+        permission_manage_orders,
+    )
+    app.permissions.add(permission_manage_orders)
```

Comment:
Wouldn't be better if you move this line to place where you're creating `App` object? Line:246